relayer/chain/ethereum/header.go

```diff
@@ -82,12 +89,45 @@ func MakeHeaderFromEthHeader(gethheader *etypes.Header, log *logrus.Entry) (*cha
 		Seal:             []types.Bytes{mixHashRLP, nonceRLP},
 	}
 
+	// Generate merkle proofs for Ethash
+	indices := ethash.Instance.GetVerificationIndices(
+		blockNumber,
+		ethash.Instance.SealHash(gethheader),
+		gethheader.Nonce.Uint64(),
+	)
+
+	proofData := make([]DoubleNodeWithMerkleProof, len(indices))
+	for i, index := range indices {
+		element, proof, err := ethashproof.CalculateProof(blockNumber, index, proofcache)
+		if err != nil {
+			return nil, err
+		}
+
+		es := element.ToUint256Array()
+		node1Bytes := make([]byte, 64)
+		node2Bytes := make([]byte, 64)
+		// Each 32 byte sequence is left-padded with 0
+		copy(node1Bytes[32-len(es[0].Bytes()):32], es[0].Bytes())
+		copy(node1Bytes[64-len(es[1].Bytes()):], es[1].Bytes())
+		copy(node2Bytes[32-len(es[2].Bytes()):32], es[2].Bytes())
+		copy(node2Bytes[64-len(es[3].Bytes()):], es[3].Bytes())
+		proofH128 := make([][16]byte, len(proof))
+		for j, pr := range proof {
+			proofH128[j] = [16]byte(pr)
+		}
+
+		proofData[i] = DoubleNodeWithMerkleProof{
+			DagNodes: [2]types.H512{
+				types.NewH512(node1Bytes),
+				types.NewH512(node2Bytes),
+			},
+			Proof: proofH128,
+		}
+	}
+
```

Comment:
Can we put this functionality into a helper function and add a test for it?

Reply:
For sure